--- a/src/Sql/AbstractSql.php
+++ b/src/Sql/AbstractSql.php
@@ -415,8 +415,23 @@
         if ($column === null) {
             return 'NULL';
         }
+
+        /**
+         * Fix for spaces on column names.
+         * @see https://github.com/laminas/laminas-db/issues/84
+         */
+        $startsWithNumber = preg_match('/^[0-9]+/', $column);
+        $hasSpaceOrDash = preg_match('/^(.+)(\s|-)+(.+)$/i', $column);
+        if ($isIdentifier) {
+            if ($startsWithNumber || $hasSpaceOrDash) {
+                $column = $platform->quoteIdentifier($column);
+            } else {
+                $column = $platform->quoteIdentifierInFragment($column);
+            }
+        }
+
         return $isIdentifier
-                ? $fromTable . $platform->quoteIdentifierInFragment($column)
+            ? $fromTable . $column
                 : $platform->quoteValue($column);
     }

